### PR TITLE
Create a master Plan doc

### PR DESCRIPTION
**Pull request by prabhatkumarthakur**
The master plan doc should contain all the process and areas of testing DOP

#### Versions:

```
$ Kubernetes version
$ Kubernetes platform
$ kubectl version

```

### CODE REVIEW

**Comment by ajeshbaby** [comment]
Capturing the items to be part of the Testplan.
- List of things to be tested
- List of items that will not be tested
- List of quality characteristics that will be tested 
- List of product quality characteristics that will not be tested
- Test execution cycles
- Test budgets according to project or overall operational budget
- Testing schedule
- Correlation between testing cycles and release plan
- Interrelation between other departments and testing team
- Scope of each test item
- Predefined entry, continuation and exit conditions for each test level
- Risks associated with test project
- Management and control of testing
- Team members responsible for each test level
- Inputs and outputs for each test level

File: master_plan/master_test_plan.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+### 3.0 Scope:
+
+#### General:
+There are different features and functionality of DOP that needed to be tested. Listing them below.
+a. Authentication.                        
+b. Browser                          
+c. Profile                                         
+d. connect cluster                                                                      
+e. Dashboard                                                                                 
+f. Scope
```
Comment (ajeshbaby) [inline master_plan/master_test_plan.md]: Topology. (To avoid ambiguity)

File: master_plan/master_test_plan.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+f. Verify that self connected cluster are shown in Administrator account.                                                   
+g. Verify the cluster connect and disconnect functionality.                                                          
+h. Verify all the dashboards are showing the graphs or data.                                        
+i. Metrics are shown for volumes and pools.                                                        
+j. Alerts are getting generated at cluster level.                                                          
+k. Verify the teaming functionality.                                                                                             
+l. Verify the dmaas functionality. Cover all the 3 providers (GCP,AWS and Minio)                                   
+m. verify the OpenEBS installation functionality.                                                               
+n. Verify the OpenEBS upgrade functionality.          
+
```
Comment (ajeshbaby) [inline master_plan/master_test_plan.md]: One more section Functional Testing.

Comment (ajeshbaby) [reply master_plan/master_test_plan.md]: Have separate md for each functional part. 

- Authentication

- Cluster connect

- Dashboard etc....

File: master_plan/master_test_plan.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+### 6.0 Features to be tested:
+a. OpenEBS installation basic and Advance.                
+b. Teaming.                                                  
+c. Dmaas.                                                           
+d. OpenEBS upgrade.                                                                       
+f. Control Plane upgrade.                                                           
+
+### 7.0 Features not to be tested:
+a. AD authentication.                                                                          
+   Reason: We don't have configure AD server in cloud.
```
Comment (ajeshbaby) [inline master_plan/master_test_plan.md]: AD authentication should be tested, Priority will be lower for now

File: master_plan/Functional_testing/Authentication/tcid-aula01-administrator can login/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+## Prerequisites:
```
Comment (ajeshbaby) [inline master_plan/Functional_testing/Authentication/tcid-aula01-administrator can login/README.md]: Name this as tcid-au1a01.md

Comment (ajeshbaby) [reply master_plan/Functional_testing/Authentication/tcid-aula01-administrator can login/README.md]: Structure could be of the directory
Testpaln -- Masterplan.md
 Testplan/subfeatures1/subfeature1.md <Details of feature and approach>
 Testplan/subfeature1/tcid1.md <Details of actual TC>
 Testplan/subfeature1/tcid2.md <Details of actual TC>
 Testplan/subfeatures2/subfeature2.md <Details of feature and approach>
 Testplan/subfeature2/tcid1.md <Details of actual TC>
 Testplan/subfeature2/tcid2.md <Details of actual TC> 
All TCID should be unique

**Comment by ajeshbaby** [comment]
@prabhatkumarthakur  can you close this as we are following a different approach now.

**Comment by prabhatkumarthakur** [comment]
Closing this PR as we are tracking it in a different way.

**Pull request closed without merging**